cfspopcon/formulas/separatrix_conditions/separatrix_operational_space/shared.py

```diff
@@ -0,0 +1,138 @@
+"""Shared functions used for computing the separatrix operational space."""
+import numpy as np
+
+from ....algorithm_class import Algorithm
+from ....unit_handling import Quantity, Unitfull, ureg, wraps_ufunc
+from ...metrics.larmor_radius import calc_larmor_radius
+
+
+def calc_ideal_MHD_wavenumber(beta_e: Unitfull, epsilon_hat: Unitfull, omega_B: Unitfull, tau_i: Unitfull, alpha_t: Unitfull) -> Unitfull:
+    """Calculate k_ideal, which gives the spatial scale of ideal MHD modes.
+
+    Equation G.3 from :cite:`Eich_2021`.
+
+    N.b. G.3 is written in terms of beta_hat = beta_e * epsilon_hat
+    where epsilon_hat = (q R / electron_pressure_decay_length)**2
+    """
+    return np.sqrt(beta_e * epsilon_hat * omega_B**1.5 * (1.0 + tau_i) / alpha_t)
+
+
+def calc_resistive_ballooning_wavenumber(critical_alpha_MHD: Unitfull, alpha_t: Unitfull, omega_B: Unitfull) -> Unitfull:
+    """Calculate k_RBM, which gives the spatial scale of resistive ballooning modes.
+
+    Equation B.12 from :cite:`Eich_2021`
+
+    N.b. There is an extra factor of Lambda_pe multiplied on omega_B compared to
+    table 1. In Appendix A it is noted that Lambda_pe = perpendicular_decay_length / electron_pressure_decay_length = 1.
+    """
+    return np.sqrt(critical_alpha_MHD / alpha_t * np.sqrt(omega_B))
+
+
+def calc_electromagnetic_wavenumber(beta_e: Unitfull, mu: Unitfull) -> Unitfull:
+    """Calculate k_EM, which gives the spatial scale of electromagnetic effects.
+
+    Equation D.3 from :cite:`Eich_2021`.
+    """
+    return np.sqrt(beta_e / mu)
```

Comment:
Is this missing a factor 2 ? 
<img width="428" alt="image" src="https://github.com/cfs-energy/cfspopcon/assets/11351671/3bbb86e1-3a76-4e99-b90d-e6b0e07fe43e">

Reply:
Or is this because the corrected factor 2 in the below `calc_electron_beta`? 
Then this function should also have a comment mentioning that

Reply:
The latter. It's pretty confusingly written, but this is what i needed to reproduce the results in the paper.

Reply:
I've added a comment to the docstring.